internal/controller/services/monitoring/monitoring_controller_support.go

```diff
@@ -362,37 +362,34 @@ func addReplicasData(ctx context.Context, rr *odhtypes.ReconciliationRequest, me
 
 // addExportersData adds custom metrics exporters data to the template data map.
 func addExportersData(metrics *serviceApi.Metrics, templateData map[string]any) error {
-	if metrics.Exporters == nil {
-		return nil
-	}
-
+	// Always initialize to avoid template rendering failures
 	validExporters := make(map[string]interface{})
-	exporterNames := make([]string, 0, len(metrics.Exporters))
-
-	for name, configYAML := range metrics.Exporters {
-		if isReservedExporterName(name) {
-			return fmt.Errorf("exporter name '%s' is reserved and cannot be used", name)
-		}
+	exporterNames := make([]string, 0)
 
-		// Trim and parse YAML configuration string
-		cfgStr := strings.TrimSpace(configYAML)
-		var cfg interface{}
-		if err := yaml.Unmarshal([]byte(cfgStr), &cfg); err != nil {
-			return fmt.Errorf("invalid YAML configuration for exporter '%s': %w", name, err)
+	if metrics.Exporters != nil {
+		// Reuse the existing validateExporters function that handles runtime.RawExtension
+		validatedExporters, err := validateExporters(metrics.Exporters)
+		if err != nil {
+			return err
 		}
 
-		// Require a mapping/object to avoid invalid collector config
-		cfgMap, ok := cfg.(map[string]interface{})
-		if !ok {
-			return fmt.Errorf("exporter '%s' configuration must be a YAML mapping/object", name)
+		// Convert the validated YAML strings to map[string]interface{} for template rendering
+		for name, configYAML := range validatedExporters {
+			var cfg map[string]interface{}
+			if err := yaml.Unmarshal([]byte(configYAML), &cfg); err != nil {
+				return fmt.Errorf("failed to unmarshal validated exporter config for '%s': %w", name, err)
+			}
+			if cfg == nil {
+				cfg = map[string]interface{}{}
+			}
+			validExporters[name] = cfg
+			exporterNames = append(exporterNames, name)
 		}
 
-		validExporters[name] = cfgMap
-		exporterNames = append(exporterNames, name)
+		// Ensure deterministic order in templates/pipelines
+		sort.Strings(exporterNames)
 	}
 
-	// Ensure deterministic order in templates/pipelines
-	sort.Strings(exporterNames)
 	templateData["MetricsExporters"] = validExporters
```

Comment:
Should we be setting nil values? would it not be more efficient to check for `metrics.Exporters == nil` and just returning nil?